### ISSUE TITLE
Integer doesn't work properly

### ISSUE THREAD

**Issue by cckk4467** [ManimCommunity/manim#1169]
here is the testing code
```python
class Test(Scene):
    def construct(self):
        text=Integer(114514)
        self.add(text)
        self.play(text.animate.set_value(23333))
        self.play(text.animate.set_value(1919810))
        self.play(text.animate.set_value(666666))
        self.play(text.animate.set_value(20210230))
        self.wait()
```
turns out that some numbers of it fail to be displayed

https://user-images.githubusercontent.com/26384746/112486579-e9e84180-8d73-11eb-898c-87a681ce214b.mp4

**Comment by kolibril13** [comment]
Seems to be an error with the animate.set_value syntax, this is working:
```py
class Test(Scene):
    def construct(self):
        self.add(Integer(114514).shift(DOWN*0.5))
        self.add(Integer(23333).shift(DOWN*1))
        self.add(Integer(1919810).shift(DOWN*1.5))
        self.add(Integer(666666).shift(DOWN*2))
        self.add(Integer(20210230).shift(DOWN*2.5))
```
![image](https://user-images.githubusercontent.com/44469195/112554189-23499d00-8dc6-11eb-9917-b5e07f05c0be.png)

**Comment by cckk4467** [comment]
> Seems to be an error with the animate.set_value syntax, this is working:
> 
> ```python
> class Test(Scene):
>     def construct(self):
>         self.add(Integer(114514).shift(DOWN*0.5))
>         self.add(Integer(23333).shift(DOWN*1))
>         self.add(Integer(1919810).shift(DOWN*1.5))
>         self.add(Integer(666666).shift(DOWN*2))
>         self.add(Integer(20210230).shift(DOWN*2.5))
> ```
> 
> ![image](https://user-images.githubusercontent.com/44469195/112554189-23499d00-8dc6-11eb-9917-b5e07f05c0be.png)

Ok...but what if I want to animate that transform a number to another number?

**Comment by kolibril13** [comment]
``` 
a=Integer(114514)
self.wait()
a.become(Integer(22245))
self.wait()
```

**Comment by JeroSQ** [comment]
There seems to be an issue with the opacity of some numbers. I've pinned it down to `match_styles()` with `family` set to `True`. When `family` is `True` it will recursively match each "new" `submobject's` style to the "old" `submobject`. One of the things it will match, is `fill_opacity`. I'm not sure why yet, but it seems that some times the `get_fill_opacities()` function returns `0` on some of the numbers, making them not show up. The opacity is grabbed from the Alpha value of the RGBA values of the `mobject`, which are returned by `get_fill_rgbas()`. I currently do not know why this happens, but I'm looking into it.

I've added some `print` statements in `vectorized_mobject.py` to see what the values of things are. You can see that the strings that don't show up have an opacity of `0`. This is true for all cases, except the first `9`(that doesn't show up but seems to have opacity `1`) so I'm really confused about that:

```
(mce) PS C:\Users\jero0\OneDrive\Documentos\Jero\Programación\Python\Manim> manim -p bug3.py Test
Manim Community v0.6.0

String: 2 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: 2 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: 3 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: , | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: 3 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: 3 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: 3 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
[05/29/21 20:01:03] INFO     Animation 0 : Partial movie file written in {'C:\\Users\\jero0\\OneDrive\\Documentos\\Jero\\Programación\\Python\\Manim\\media\\videos\\bug3\\1080p scene_file_writer.py:401                             60\\partial_movie_files\\Test\\450974505_4119951202_993979061.mp4'}
String: 1 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: , | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: 9 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: 1 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: 9 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: , | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: 8 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: 1 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: 0 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
Animation 1: _MethodAnimation(Integer):  20%|########################6                                                                                                  | 12/60 [00:00<00:01, 25.55it/s]
(process:7896): GLib-GIO-WARNING **: 20:01:04.504: Unexpectedly, UWP app `3138AweZip.AweZip_1.4.2.0_x86__ffd303wmbhcjt' (AUMId `3138AweZip.AweZip_ffd303wmbhcjt!App') supports 167 extensions but has no 
verbs
[05/29/21 20:01:05] INFO     Animation 1 : Partial movie file written in {'C:\\Users\\jero0\\OneDrive\\Documentos\\Jero\\Programación\\Python\\Manim\\media\\videos\\bug3\\1080p scene_file_writer.py:401                             60\\partial_movie_files\\Test\\2788726626_3396817513_993979061.mp4'}
String: 6 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: 6 | RGBAS: [[1. 1. 1. 0.]] OPACITY [0.]
String: 6 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: 6 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: , | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: , | RGBAS: [[1. 1. 1. 0.]] OPACITY [0.]
String: 6 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: 6 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: 6 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
RGBAS: [[1. 1. 1. 0.]] OPACITY [0.]
RGBAS: [[1. 1. 1. 0.]] OPACITY [0.]
[05/29/21 20:01:08] INFO     Animation 2 : Partial movie file written in {'C:\\Users\\jero0\\OneDrive\\Documentos\\Jero\\Programación\\Python\\Manim\\media\\videos\\bug3\\1080p scene_file_writer.py:401                             60\\partial_movie_files\\Test\\2788726626_1432049237_1413979930.mp4'}
String: 2 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: 0 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: , | RGBAS: [[1. 1. 1. 0.]] OPACITY [0.]
String: 2 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: 1 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: 0 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: , | RGBAS: [[1. 1. 1. 0.]] OPACITY [0.]
String: 2 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: 3 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
String: 0 | RGBAS: [[1. 1. 1. 1.]] OPACITY [1.]
```

**Comment by JeroSQ** [comment]
I don't really know where to go from here, so I'll leave the facts I've gathered in case someone else wants to take a look:

- As expected, this issue is also present with `DecimalNumber`, since its the parent class of `Integer`.
- Digits starts to disappear when the number of digits changes, but it doesn't happen every time. Seems to happen both when the n° of digits increases or decreases, but again, it doesn't happen 100% of the time.
- Setting `family=False` in `match_style()` in `set_value()` would solve it, but said change makes the styling to be essentially lost when changing the value.
- I think it has something to do with the `make_even()` function. Fiddling around makes some numbers to reappear but some others to disappear. I tried replacing `make_even()` with `make_even_by_cycling()` or just removing it but it didn't solve the issue, it just changed the output.
